### ISSUE TITLE
Lored item trading bug

### ISSUE THREAD

**Issue by blablubbabcDEV** [Shopkeepers/Shopkeepers#251]
**Migrated from:** https://dev.bukkit.org/projects/shopkeepers/issues/246

**Originally posted by Thyrin (Aug 1, 2014):** 

What steps will reproduce the problem?
1. Install a 1.7.10 build and Shopkeepers
2. Create a shopkeeper with a lored item as ingredient
3. Trade with an item of the same type but with different lore.What is the expected output? What do you see instead?
I expect the trade to fail, but it works instead.What version of the product are you using?
I tested with: Spigot 1590 and 1584, Bukkit 3105 and 3100
Shopkeepers: Latest from source, 1.22, and 1.18Please provide any additional information below.
I have done some testing from the source code. From what I've found the itemstack similarity check woks fine, and it does actually do event.setCancelled(true) but then the actual event in-game does not get cancelled. This could indicate a problem with Bukkit/Mojang but I've not been able to track down where exactly. I'm hoping you're able to have more luck, because trading lored items is a crucial part of my server's economy.If there is anything I can do to help let me know, I know javas.

**Comment by blablubbabcDEV** [comment]
**Originally commented by Thyrin (Aug 1, 2014):** 

For the record: this process fails on 1.7.10, but works fine on 1.7.9

**Comment by blablubbabcDEV** [comment]
**Originally commented by blablubbabc (Aug 1, 2014):** 

I just tested this on craftbukkit build 3100 (1.7.10), with shopkeepers v1.23 (the latest from source) and it was working fine, just like expected..

For the test I summoned 2 diamond swords, one with lore "test" the other with lore "test2".

I then created an admin shopkeeper and setup a trade: the diamond sword with lore "test" for 1 emerald.

If I then put in the diamond sword with the lore "test2" the trade failed, just like always..
Also check that you don't have setup another trade which accepts the item with the different lore in the shopkeeper you are testing with.
